File: Laborator4/Laborator4_RecyclerView1/app/src/main/java/com/example/laborator4_recyclerview1/ProductDetailActivity.java
package com.example.laborator4_recyclerview1;


import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.ImageView;
import android.widget.TextView;

public class ProductDetailActivity extends AppCompatActivity {

    ImageView imageViewDetail;
    TextView textViewName, textViewInfo;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_product_detail);

        imageViewDetail = findViewById(R.id.imageViewDetail);
        textViewName = findViewById(R.id.textViewDetailName);
        textViewInfo = findViewById(R.id.textViewDetailInfo);

        // Preluăm datele trimise din intent
        Bundle extras = getIntent().getExtras();
        if (extras != null) {
            String name = extras.getString("name");
            String details = extras.getString("details");
            int imageResId = extras.getInt("imageResId");

            textViewName.setText(name);
            textViewInfo.setText(details);
            imageViewDetail.setImageResource(imageResId);
        }
    }
}

File: Laborator5/Kotlin_RecylerView_1/app/src/main/java/com/example/kotlin_recylerview_1/view/ProductDetailActivity.kt
package com.example.kotlin_recylerview_1.view

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ImageView
import android.widget.TextView
import com.example.kotlin_recylerview_1.R

class ProductDetailActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_product_detail)

        val imageView: ImageView = findViewById(R.id.imageViewDetail)
        val textName: TextView = findViewById(R.id.textViewDetailName)
        val textDetails: TextView = findViewById(R.id.textViewDetailInfo)

        val name = intent.getStringExtra("name")
        val details = intent.getStringExtra("details")
        val imageResId = intent.getIntExtra("imageResId", 0)

        textName.text = name
        textDetails.text = details
        imageView.setImageResource(imageResId)
    }
}
